contracts/libraries/VaultLifecycleWithSwap.sol

```diff
@@ -0,0 +1,745 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Vault} from "./Vault.sol";
+import {ShareMath} from "./ShareMath.sol";
+import {IStrikeSelection} from "../interfaces/IRibbon.sol";
+import {GnosisAuction} from "./GnosisAuction.sol";
+import {
+    IOtokenFactory,
+    IOtoken,
+    IController,
+    GammaTypes
+} from "../interfaces/GammaInterface.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {SupportsNonCompliantERC20} from "./SupportsNonCompliantERC20.sol";
+
+library VaultLifecycleWithSwap {
+    using SafeMath for uint256;
+    using SupportsNonCompliantERC20 for IERC20;
+
+    struct CommitParams {
+        address OTOKEN_FACTORY;
+        address USDC;
+        address currentOption;
+        uint256 delay;
+        uint16 lastStrikeOverrideRound;
+        uint256 overriddenStrikePrice;
+    }
+
+    /**
+     * @notice Sets the next option the vault will be shorting, and calculates its premium for the auction
+     * @param strikeSelection is the address of the contract with strike selection logic
+     * @param optionsPremiumPricer is the address of the contract with the
+       black-scholes premium calculation logic
+     * @param premiumDiscount is the vault's discount applied to the premium
+     * @param commitParams is the struct with details on previous option and strike selection details
+     * @param vaultParams is the struct with vault general data
+     * @param vaultState is the struct with vault accounting state
+     * @return otokenAddress is the address of the new option
+     * @return premium is the premium of the new option
+     * @return strikePrice is the strike price of the new option
+     * @return delta is the delta of the new option
+     */
+    function commit(
```

Comment:
Suggestion, we can rename this to `commitToNextOption`

Reply:
Changed to `commitNextOption`.